### PR TITLE
support azure vmss

### PR DESCRIPTION
**Pull request by luthermonson**

### CODE REVIEW

File: cloud-provider.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 AZURE_META_URL="http://169.254.169.254/metadata/instance/compute"
+AZURE_META_API_VERSION="2019-11-01"
```
Comment (superseb) [inline cloud-provider.sh]:
```
The version 2019-11-01 is currently getting deployed and may not be available in all regions.
```

Is this going to bite us? https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service

Comment (deniseschannon) [reply cloud-provider.sh]: Maybe we should use 2019-08-15 as that's in all regions?

File: cloud-provider.sh [hunk 2]
```diff
@@ -38,48 +39,65 @@ set_azure_config() {
...
   fi
 
-  vmcmd="vm" # default is "standard" and needs "vm" cmd
   if [ "$az_vm_type" = "vmss" ]; then
-    vmcmd="vmss" # scale set cmd
-  fi
+    # vmss
+    local az_vm_scale_set_name=$(curl  -s -H Metadata:true "${AZURE_META_URL}/vmScaleSetName?api-version=${AZURE_META_API_VERSION}&format=text")
+    local az_vm_instance_id=$(az vmss list-instances -g ${az_resources_group} --name ${az_vm_scale_set_name} --query "[?name=='${az_vm_name}'].instanceId" --output tsv)
+    local az_vm_nic=$(az vmss nic list -g ${az_resources_group} --vmss-name ${az_vm_scale_set_name} --output json --query [0].id)
```
Comment (superseb) [inline cloud-provider.sh]: Is there a specific reason for `.id` and the post processing vs `query [0].name`?

File: cloud-provider.sh (hunk 1 above)
Comment (luthermonson) [reply cloud-provider.sh]: ya taht's fine, also why i made it a param

Comment (deniseschannon) [reply cloud-provider.sh]: Let's default to `2019-08-15`.

Comment (luthermonson) [reply cloud-provider.sh]: 2019-08-15 it is

File: cloud-provider.sh (hunk 2 above)
Comment (luthermonson) [reply cloud-provider.sh]: its possible that was added in later api versions and not inthe 2017 one this was built on.. im going to update it cause it's much cleaner.

**Comment by luthermonson** [comment]
https://github.com/rancher/rancher/issues/26293

**Pull request merged**